src/exe/cimbar_recv/recv.cpp

```diff
@@ -136,41 +130,29 @@ int main(int argc, char** argv)
 			continue;
 		}
 
-		cv::Mat img = mat.clone();
+		cv::UMat img = mat.getUMat(cv::ACCESS_RW).clone();
 		cv::cvtColor(mat, mat, cv::COLOR_BGR2RGB);
 
 		// draw some stats on mat?
 		window.show(mat, 0);
 
-		// extract, decode, etc
-		int bytes = cimbard_scan_extract_decode(img.data, img.cols, img.rows, 3, bufspace.data(), bufspace.size());
-		if (bytes <= 0)
-			continue;
-
-		if (bytes % chunkSize != 0)
+		// extract
+		bool shouldPreprocess = false;
+		int res = ext.extract(img, img);
```

Comment:
In #134, I updated the cimbar_recv app to use the new C api rather than the C++ classes directly. The android app uses the C++ classes though, and probably will continue to do so. So... for now, we'll just have 2 different test apps. cimbar_recv2 will use the C api, cimbar_recv the C++ classes.

Reply:
I do have some hope that at some point I'll be happy enough with the C api to switch to using it everywhere, but it's definitely not there yet.